pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -418,16 +444,17 @@ protected BrokerResponse doHandleRequest(long requestId, String query, SqlNodeAn
     String realtimeTableName = routeInfo.getRealtimeTableName();
     TableConfig offlineTableConfig = routeInfo.getOfflineTableConfig();
     TableConfig realtimeTableConfig = routeInfo.getRealtimeTableConfig();
+    QueryConfig offlineTableQueryConfig = routeInfo.getOfflineTableQueryConfig();
+    QueryConfig realtimeTableQueryConfig = routeInfo.getRealtimeTableQueryConfig();
     TimeBoundaryInfo timeBoundaryInfo = routeInfo.getTimeBoundaryInfo();
 
-    HandlerContext handlerContext = getHandlerContext(offlineTableConfig, realtimeTableConfig);
+    HandlerContext handlerContext = getHandlerContext(offlineTableQueryConfig, realtimeTableQueryConfig);
     validateGroovyScript(serverPinotQuery, handlerContext._disableGroovy);
     if (handlerContext._useApproximateFunction) {
       handleApproximateFunctionOverride(serverPinotQuery);
     }
 
     // Validate QPS quota
-    String database = DatabaseUtils.extractDatabaseFromFullyQualifiedTableName(tableName);
     if (!_queryQuotaManager.acquireDatabase(database)) {
       String errorMessage =
           String.format("Request %d: %s exceeds query quota for database: %s", requestId, query, database);
```

Comment:
Check the `requestContext.setOfflineServerTenant` when `offlineBrokerRequest` is prepared. The `getServerTenant` does not handle the logical tables. See if any changes are needed.

Same for the realtime time table.